lms/templates/header/navbar-not-authenticated.html

```diff
@@ -48,7 +48,7 @@
   <div class="secondary">
     <div>
       % if allows_login:
-        % if allow_public_account_creation:
+        % if allow_public_account_creation and not disable_register_button:
```

Comment:
#NIT The `not` condition makes this logic a bit difficult to read. How about updating the flag to be `enable_register_button_on_nav`. Then make that always `True`? If that's too much work, please feel free to ignore here.

Reply:
Also too big of a possibility of missing one